client/finality-grandpa/src/communication/mod.rs

```diff
@@ -77,13 +77,13 @@ pub mod grandpa_protocol_name {
 	/// Name of the notifications protocol used by GRANDPA.
```

Comment:
Was the previous name deployed? Should we include it in `LEGACY_NAMES` to provide backward compatibility?

Reply:
Sadly yes.

Reply:
I'm a bit lost in your question, but yes there's a handshake. We try whether `protocol_name` is supported, and if not we try `LEGACY_NAMES`. It's only if neither is supported that we give up.